src/AddStep.js

```diff
@@ -25,10 +25,10 @@ function AddStep(_sequencer, image, name, o) {
       options: o
     };
     var UI = _sequencer.events;
-    this.getStep = function getStep(offset) {
-      return _sequencer.images[image].steps.slice(offset - 1)[0];
+    getStep = function getStep(input, offset) {
```

Comment:
wait, why does this have to be global? So that it can be accessed from draw()? Yeah, i think we probably don't want to do this... it can get very messy. Let me think a bit more about it!

Reply:
@jywarren Actually The idea here is that getStep() is a function on sequencer itself and then what we pass it is a context and and an offset

Reply:
`getStep(context,offset)`

Reply:
@jywarren the other solution we did before this essentially made it global as well!! that is why the data was being shared across all steps. If we want to call a function without using `something.getStep()` then either getStep has to be defined locally inside of the module or we have to make it global, I read a lot about this and we cannot inject local variables inside of a function, all we can remap is the `this` pointer